# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.cpp
# Commit: 6cbb442f4772
# Full Hash: 6cbb442f477260469cab9f1d81e33fcaede82029
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - FetchService integration for PFetch. r=dom-worker-reviewers,jesup
#   
#   Depends on D142436
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142437
# ==============================================================================

diff -r dbdca4661a35 -r 6cbb442f4772 dom/fetch/InternalResponse.cpp
--- a/dom/fetch/InternalResponse.cpp	Tue Jan 17 15:13:45 2023 +0000
+++ b/dom/fetch/InternalResponse.cpp	Tue Jan 17 15:13:46 2023 +0000
@@ -90,6 +90,13 @@
         aIPCResponse.metadata().principalInfo().ref()));
   }
 
+  nsAutoCString bodyBlobURISpec(aIPCResponse.metadata().bodyBlobURISpec());
+  response->SetBodyBlobURISpec(bodyBlobURISpec);
+  nsAutoString bodyLocalPath(aIPCResponse.metadata().bodyLocalPath());
+  response->SetBodyLocalPath(bodyLocalPath);
+
+  response->mCredentialsMode = aIPCResponse.metadata().credentialsMode();
+
   switch (aIPCResponse.metadata().type()) {
     case ResponseType::Basic:
       response = response->BasicResponse();
@@ -124,13 +131,17 @@
   Maybe<mozilla::ipc::PrincipalInfo> principalInfo =
       mPrincipalInfo ? Some(*mPrincipalInfo) : Nothing();
 
+  nsAutoCString bodyBlobURISpec(BodyBlobURISpec());
+  nsAutoString bodyLocalPath(BodyLocalPath());
+
   // Note: all the arguments are copied rather than moved, which would be more
   // efficient, because there's no move-friendly constructor generated.
   nsCOMPtr<nsITransportSecurityInfo> securityInfo(mChannelInfo.SecurityInfo());
   return InternalResponseMetadata(
       mType, GetUnfilteredURLList(), GetUnfilteredStatus(),
       GetUnfilteredStatusText(), headersGuard, headers, mErrorCode,
-      GetAlternativeDataType(), securityInfo, principalInfo);
+      GetAlternativeDataType(), securityInfo, principalInfo, bodyBlobURISpec,
+      bodyLocalPath, GetCredentialsMode());
 }
 
 void InternalResponse::ToChildToParentInternalResponse(
@@ -196,15 +207,16 @@
   GetUnfilteredBody(getter_AddRefs(body), &bodySize);
 
   if (body) {
-    result.body() = Some(
-        ToParentToChildStream(WrapNotNull(body), bodySize, aBackgroundParent));
+    result.body() = Some(ToParentToChildStream(
+        WrapNotNull(body), bodySize, aBackgroundParent, mSerializeAsLazy));
     result.bodySize() = bodySize;
   }
 
   nsCOMPtr<nsIInputStream> alternativeBody = TakeAlternativeBody();
   if (alternativeBody) {
-    result.alternativeBody() = Some(ToParentToChildStream(
-        WrapNotNull(alternativeBody), UNKNOWN_BODY_SIZE, aBackgroundParent));
+    result.alternativeBody() = Some(
+        ToParentToChildStream(WrapNotNull(alternativeBody), UNKNOWN_BODY_SIZE,
+                              aBackgroundParent, mSerializeAsLazy));
   }
 
   return result;